app/models/spree/image.rb

```diff
@@ -35,6 +35,12 @@ def url(size)
       Rails.application.routes.url_helpers.url_for(variant(size))
     end
 
+    def cdn_url(size)
+      return url(size) if ENV["S3_BUCKET"].nil?
+
+      variant(size).service_url
```

Comment:
It is not suggested to use this method publicly, but it should not be such an issue in our case as the images we want to display are actually public. I have not find an easier way to display the image directly from the cloud storage, this method is setting up all we need.